app/src/main/java/com/example/bigowlapp/model/Notification.java

```diff
@@ -9,8 +9,13 @@
 
     @DocumentId
     private String uId;
+    private String senderUId;
+    private String receiverUId;
+    private String groupUId;
     private String type;
-    private Timestamp time;
+    private Timestamp timeRead;
+    private Timestamp timeSend;
+    private UserScheduleResponse senderResponse;
```

Comment:
This is a Notification class, thus UserScheduleResponse shouldn't been in the "Notification" class as well as the other variables.

A Notification's purpose should only be used for notifications. I understand that this solution is used because you may have a problem with polymorphism and subclasses, but a preferable solution is to create a new class called UserScheduleNotification as a temporary solution.

This solution is much better since you aren't coupling and mixing up classes together. This will be a problem if Notifications such as friend requests or other types of requests are involved.